pkg/services/ssosettings/ssosettingsimpl/service.go

```diff
@@ -196,9 +196,11 @@ func (s *Service) Upsert(ctx context.Context, settings *models.SSOSettings) erro
 		return err
 	}
 
-	settings.Settings = overrideMaps(storedSettings.Settings, settings.Settings, secrets)
+	// make a copy of current setings for reload operation and apply overrides
+	reloadSettings := *settings
+	reloadSettings.Settings = overrideMaps(storedSettings.Settings, settings.Settings, secrets)
 
-	go s.reload(social, settings.Provider, *settings)
+	go s.reload(social, settings.Provider, reloadSettings)
```

Comment:
Thanks @mgyongyosi for helping me realize here was the source of the other data race in the tests.